spec/classes/powerdns_init_spec.rb

```diff
@@ -131,6 +131,25 @@
           it { is_expected.to contain_service('pdns').with('enable' => 'true') }
           it { is_expected.to contain_service('pdns').with('name' => authoritative_service_name) }
           it { is_expected.to contain_service('pdns').that_requires("Package[#{authoritative_package_name}]") }
+
+          if facts[:os]['family'] == 'RedHat'
+            it do
+              is_expected.to contain_file(authoritative_config).with(
+                ensure: 'file',
+                owner: 'root',
+                group: 'pdns',
+                mode: '0640',
+              ).that_comes_before('Service[pdns]')
+            end
+          else
+            it do
+              is_expected.to contain_file(authoritative_config).with(
+                ensure: 'file',
+                owner: 'root',
+                mode: '0640',
+              ).that_comes_before('Service[pdns]')
+            end
+          end
```

Comment:
Perhaps keep this to 2 separate assertions?
```suggestion
          if facts[:os]['family'] == 'RedHat'
            it { is_expected.to contain_file(authoritative_config).with(group: 'pdns') }
          end

          it do
            is_expected.to contain_file(authoritative_config).with(
              ensure: 'file',
              owner: 'root',
              mode: '0640',
            ).that_comes_before('Service[pdns]')
          end
```

Reply:
Makes perfect sense - done!